tracking/src/main/java/org/hps/recon/tracking/kalman/KalmanPatRecDriver.java

```diff
@@ -502,32 +522,29 @@ private ArrayList<KalTrack>[] prepareTrackCollections(EventHeader event, List<Tr
 
                 //Add the TrackResiduals
                 List<Integer> layers    = new ArrayList<Integer>();
-                List<Double> residuals  = new ArrayList<Double>();
+		List<Double> residuals  = new ArrayList<Double>();
                 List<Float> sigmas      = new ArrayList<Float>(); 
-		        List<Integer> layersInt = new ArrayList<Integer>();
-		        List<Double>  intersect = new ArrayList<Double>();
+		List<Integer> layersInt = new ArrayList<Integer>();
+		List<Double>  intersect = new ArrayList<Double>();
                 List<Float>   sigmasInt = new ArrayList<Float>();
                 int uindex = 0;
                 int vindex = 1;
                 int windex = 2;
-		//...loop over clusters and save millipedID to residuals
-                for (GBLStripClusterData clstr : clstrs) {
-                    Pair<Double,Double> res_and_sigma = kTk.unbiasedResidualMillipede(clstr.getId());
+                for (int ilay = 0; ilay<14; ilay++) {
```

Comment:
Leave this as loop over cluster data and the layer as the millipede ID (clstr.getId())

Reply:
Fixed